packages/dart_mappable_builder/lib/src/generators/enum_mapper_generator.dart

```diff
@@ -7,6 +7,8 @@ class EnumMapperGenerator extends MapperGenerator<TargetEnumMapperElement> {
 
   @override
   Future<String> generate() async {
+    final extensionReturnType = element.hasAllStringValues ? 'String' : 'dynamic';
```

Comment:
The `hasAllStringValues` currently does not check custom annotated values and would be false in the case of the issue.

Reply:
I will check to change the implementation